### PR TITLE
Add type definitions for package ringbufferjs

### PR DESCRIPTION
**Pull request by amchelle**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/ringbufferjs/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// Type definitions for ringbufferjs 1.1
+// Project: https://github.com/janogonzalez/ringbufferjs#readme
+// Definitions by: Amchelle Clendenin <https://github.com/amchelle>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export default class RingBuffer<T> {
```
Comment (uniqueiniquity) [inline types/ringbufferjs/index.d.ts]: It seems like the [underlying library](https://unpkg.com/ringbufferjs@1.1.0/index.js) is not using `export default`, so this .d.ts file should not either. Instead, you should follow the appropriate template from [the TypeScript handbook](http://www.typescriptlang.org/docs/handbook/declaration-files/templates.html). Note that if you want to use `import name from 'package';` in your consuming code, you can use the `esModuleInterop` compiler setting.

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

File: types/ringbufferjs/index.d.ts (hunk 1 above)
Comment (amchelle) [reply types/ringbufferjs/index.d.ts]: Ok thanks for the tip! Let me take a look

Comment (amchelle) [reply types/ringbufferjs/index.d.ts]: I removed the `default export`, please let me know if there is anything else

Comment (uniqueiniquity) [reply types/ringbufferjs/index.d.ts]: I think you actually want to use [this template](http://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-class-d-ts.html).

The way you have the declaration file right now, and shown in the tests, will fail at runtime.

Comment (amchelle) [reply types/ringbufferjs/index.d.ts]: Thank you for pointing me to the right template! I think I found the required reading on `esModuleInterop`. Third time's the charm

**Review by uniqueiniquity — APPROVED** [review]

**Comment by uniqueiniquity** [comment]
Congratulations on your first DefinitelyTyped contribution!
Thank you for being a part of the community!

**Pull request merged**